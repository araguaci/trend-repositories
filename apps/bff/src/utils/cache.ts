const LRU = require('lru-cache');

export let cache = null;

export const initCache = () => {
  const options = {
    max: 500,
    ttl: 86400 * 1000, // 1 day
  };

  cache = new LRU(options);
};
