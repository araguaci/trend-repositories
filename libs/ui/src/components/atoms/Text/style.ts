import styled from 'styled-components';
import { theme } from '@trend-repositories/ui';

export interface TextProps {
  size: keyof typeof theme.sizes;
  elementType?: 'span' | 'p';
}

export const StyledText = styled.p<Pick<TextProps, 'size'>>`
  font-size: ${(props) => props.theme.sizes[props.size]};
  color: ${(props) => props.theme.colors.grey_1};
`;
