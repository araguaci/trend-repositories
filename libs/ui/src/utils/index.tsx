/* istanbul ignore file */
import { ThemeProvider } from 'styled-components';
import theme from '../theme';
import { ReactNode } from 'react';
import { render, act } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { MockedResponse } from '@apollo/client/testing/core';
import { InMemoryCache } from '@apollo/client/cache';

export const renderWithProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const renderWithMockProvider = (
  node: ReactNode,
  cache: InMemoryCache,
  mocks: ReadonlyArray<MockedResponse>
) => {
  return render(
    <ThemeProvider theme={theme}>
      <MockedProvider cache={cache} mocks={mocks}>
        {node}
      </MockedProvider>
    </ThemeProvider>
  );
};

export const renderWithThemeProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const waitForResponse = async () =>
  await act(async () => await new Promise((resolve) => setTimeout(resolve, 0)));
export * from '@testing-library/react';
