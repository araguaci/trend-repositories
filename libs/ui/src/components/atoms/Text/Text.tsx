import { StyledText, TextProps } from './style';
import { FC } from 'react';

const Text: FC<TextProps> = ({ children, size, elementType = 'span' }) => {
  return (
    <StyledText as={elementType} size={size}>
      {children}
    </StyledText>
  );
};

export default Text;
