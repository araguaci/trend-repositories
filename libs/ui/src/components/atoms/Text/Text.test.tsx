import Text from './Text';
import { renderWithProvider } from '../../../utils';

describe('Text Component', () => {
  it('should be rendered correctly', () => {
    const { asFragment } = renderWithProvider(
      <Text size="small">linguistic</Text>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be rendered as a paragraph', () => {
    const { asFragment } = renderWithProvider(
      <Text elementType="p" size="small">
        linguistic
      </Text>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
