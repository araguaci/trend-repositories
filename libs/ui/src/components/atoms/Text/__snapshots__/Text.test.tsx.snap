// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component should be rendered as a paragraph 1`] = `
<DocumentFragment>
  <p
    class="style__StyledText-sc-1aawx67-0 kEyWhB"
  >
    linguistic
  </p>
</DocumentFragment>
`;

exports[`Text Component should be rendered correctly 1`] = `
<DocumentFragment>
  <span
    class="style__StyledText-sc-1aawx67-0 kEyWhB"
  >
    linguistic
  </span>
</DocumentFragment>
`;
